src/VisualStudio/TestUtilities/IntegrationHelper.cs

```diff
@@ -0,0 +1,209 @@
+                IntegrationLog.Current.WriteLine($"    {e}");
+            }
+            catch (InvalidOperationException e)
+            {
+                IntegrationLog.Current.Warning("The caller attempted to kill a process which either does not exist or has already been terminated.");
+                IntegrationLog.Current.WriteLine($"    {e}");
+            }
+        }
+
+        public static void KillProcess(string processName)
+        {
+            foreach (var process in Process.GetProcessesByName(processName))
+            {
+                KillProcess(processName);
+            }
+        }
+
+        public static string LocateFile(string fileName, string downloadUrl = "")
+        {
+            var filePath = Path.Combine(Environment.CurrentDirectory, fileName);
+
+            if (File.Exists(filePath))
+            {
+                return filePath;
+            }
+
+            filePath = Path.Combine(typeof(IntegrationHelper).Assembly.Location, fileName);
+
+            if (File.Exists(filePath))
+            {
+                return filePath;
+            }
+
+            if (!string.IsNullOrWhiteSpace(downloadUrl))
+            {
+                filePath = Path.Combine(Environment.CurrentDirectory, fileName);
+
+                using (var webClient = new WebClient())
+                {
+                    webClient.DownloadFile("https://live.sysinternals.com/procdump.exe", filePath);
+                }
+
+                return filePath;
+            }
+
+            throw new FileNotFoundException($"Unable to locate the specified file. (FileName: '{fileName}')", fileName);
+        }
+
+        public static void Sleep(int sleepDelay = DefaultSleepDelay)
+        {
+            Thread.Sleep(sleepDelay);
+        }
+
+        public static bool TryDeleteDirectoryRecursively(string path)
+        {
+            try
+            {
+                DeleteDirectoryRecursively(path);
+                return true;
+            }
+            catch (Exception e)
+            {
+                IntegrationLog.Current.Warning($"Failed to recursively delete the specified directory. (Name: '{path}')");
+                IntegrationLog.Current.WriteLine($"    {e}");
+                return false;
+            }
+        }
+
+        public static void WaitForResult<T>(Func<T> action, T expectedResult, int timeout = DefaultTimeout, int sleepDelay = DefaultSleepDelay)
+        {
+            long startTime = CurrentTickCount;
+
+            while (!action().Equals(expectedResult))
+            {
+                if ((CurrentTickCount - startTime) > timeout)
+                {
+                    throw new TimeoutException($"Timed out while waiting for an action to return '{expectedResult}'. (Timeout: '{timeout}'; Sleep Delay: '{sleepDelay}')");
+                }
+
+                Sleep(sleepDelay);
```

Comment:
Please remove this call.  This pattern of waiting and sleeping is a constant source of test instability.